annot8-components-properties/src/main/java/io/annot8/components/properties/processors/TableToProperties.java

```diff
@@ -0,0 +1,179 @@
+                        }
+                      }));
+
+      rows.forEach(
+          (k, r) -> {
+            if (k.isBlank()) return;
+
+            Object value;
+            if (r.size() == 1) {
+              value = r.get(0).getValueAt(1).orElse(null);
+            } else {
+              if (settings.getJoin() == null) {
+                value =
+                    r.stream()
+                        .map(row -> row.getValueAt(1).orElse(null))
+                        .filter(Objects::nonNull)
+                        .collect(Collectors.toList());
+              } else {
+                value =
+                    r.stream()
+                        .map(row -> row.getValueAt(1).orElse(null))
+                        .filter(Objects::nonNull)
+                        .map(Object::toString)
+                        .filter(s -> !s.isEmpty())
+                        .collect(Collectors.joining(settings.getJoin()));
+              }
+            }
+
+            if (value == null) return;
+
+            if (value instanceof String && ((String) value).isEmpty()) return;
+
+            if (value instanceof List && ((List<?>) value).isEmpty()) return;
+
+            item.getProperties().set(k, value);
+          });
+
+      if (settings.isDiscardTable()) {
+        item.getContents(TableContent.class)
+            .filter(tc -> tc.getData().getColumnCount() == 2)
+            .forEach(tc -> item.removeContent(tc.getId()));
+      }
+
+      return ProcessorResponse.ok();
+    }
+
+    public static String toCamelCase(String name) {
+      String lcase =
+          name.replaceAll("/", " ").replaceAll("[^a-zA-Z0-9 ]", "").strip().toLowerCase();
+
+      String[] s = lcase.split(" ");
+
+      StringJoiner sj = new StringJoiner("");
+      sj.add(s[0]);
+
+      for (int i = 1; i < s.length; i++) {
+        if (s[i].length() > 0) {
+          sj.add(s[i].substring(0, 1).toUpperCase());
+          sj.add(s[i].substring(1));
+        }
+      }
+
+      return sj.toString();
+    }
+  }
+
+  public static class Settings implements io.annot8.api.settings.Settings {
+    private StringTransformation propertyNameTransformation = StringTransformation.CAMEL_CASE;
+    private String join = null;
+    private boolean discardTable = false;
+
+    @Override
+    public boolean validate() {
+      return propertyNameTransformation != null;
+    }
+
+    @Description(
+        "What transformation should be applied to property values prior to creation of the new property")
+    public StringTransformation getPropertyNameTransformation() {
+      return propertyNameTransformation;
+    }
+
+    public void setPropertyNameTransformation(StringTransformation propertyNameTransformation) {
+      this.propertyNameTransformation = propertyNameTransformation;
+    }
+
+    @Description(
+        "If null, then any values with the same key will be returned as a list. Otherwise, then values will be converted to strings and joint with this value.")
```

Comment:
```suggestion
        "If null, then any values with the same key will be returned as a list. Otherwise, then values will be converted to strings and joined with this value.")
```